src/frontend/contexts/MetricsContext.tsx

```diff
@@ -0,0 +1,37 @@
+import * as React from 'react';
+import {AppDiagnosticMetrics} from '../metrics/AppDiagnosticMetrics';
+import {DeviceDiagnosticMetrics} from '../metrics/DeviceDiagnosticMetrics';
+
+export type MetricsContextType = {
+  appMetrics: AppDiagnosticMetrics;
+  deviceMetrics: DeviceDiagnosticMetrics;
+};
+
+const MetricsContext = React.createContext<MetricsContextType | undefined>(
+  undefined,
+);
+
+export type MetricsProviderProps = {
+  children?: React.ReactNode;
+};
+
+export const MetricsProvider = ({
+  children,
+}: MetricsProviderProps): JSX.Element => {
+  const appMetrics = new AppDiagnosticMetrics();
+  const deviceMetrics = new DeviceDiagnosticMetrics();
```

Comment:
these initializations shouldn't live in the component render body because then they'll get re-instantiated whenever this component re-renders (which will happen due to our architecture + React context limitations). Either:

1. Instantiate "globally" in App.tsx. Then you can do something similar to what we do with other instances like the api and message port i.e. provide the instances as [props to the `AppProviders` component](https://github.com/digidem/comapeo-mobile/blob/ebaf02b9e8ea8e4806a87ed364e5ed77d416300d/src/frontend/contexts/AppProviders.tsx#L27-L32) and pass those props to this `MetricsProvider` component (similarly to what we do [here](https://github.com/digidem/comapeo-mobile/blob/ebaf02b9e8ea8e4806a87ed364e5ed77d416300d/src/frontend/contexts/AppProviders.tsx#L49-L51))

2. Move these outside the component

I think 1 is preferable because it ensures that they're instantiated as soon as possible, which I think is desirable?